api/src/main/java/uk/gov/dvsa/mot/trade/read/core/TradeReadServiceDatabase.java

```diff
@@ -104,16 +106,28 @@ public TradeReadServiceDatabase(MotTestReadService motTestReadService, VehicleRe
 
         List<Vehicle> vehicles = vehicleReadService.findByRegistration(registration);
 
+        if (vehicles.isEmpty()) {
```

Comment:
Would prefer a null check here too, even though not essential here because we can see the response (at the moment) will never be null.

E.g.

`if (CollectionUtils.isNullOrEmpty(vehicles)) {
    ...
}`